File: app/src/main/java/com/sanousun/mdweather/ui/fragment/AboutFragment.java
package com.sanousun.mdweather.ui.fragment;

import android.app.AlertDialog;
import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;

import com.sanousun.mdweather.R;

import de.psdev.licensesdialog.LicensesDialog;

public class AboutFragment extends PreferenceFragment
        implements Preference.OnPreferenceClickListener {

    public static AboutFragment newInstance() {
        return new AboutFragment();
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        addPreferencesFromResource(R.xml.preference);
        findPreference("license").setOnPreferenceClickListener(this);
        findPreference("email").setOnPreferenceClickListener(this);
        findPreference("github").setOnPreferenceClickListener(this);
    }

    @Override
    public boolean onPreferenceClick(final Preference preference) {
        final ClipboardManager cm =
                (ClipboardManager) getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
        switch (preference.getKey()) {
            case "license":
                new LicensesDialog.Builder(getActivity()).
                        setNotices(R.raw.notices).setIncludeOwnLicense(true).
                        build().showAppCompat();
                break;
            default:
                new AlertDialog.Builder(getActivity()).
                        setMessage("复制到剪贴板？").
                        setPositiveButton("确定", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                ClipData myClip;
                                CharSequence text = preference.getSummary();
                                myClip = ClipData.newPlainText("text", text);
                                cm.setPrimaryClip(myClip);
                            }
                        }).
                        setNegativeButton("取消", new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                dialogInterface.cancel();
                            }
                        }).show();
                break;
        }
        return true;

    }
}

File: app/src/main/java/com/sanousun/mdweather/ui/fragment/AboutFragment.kt
package com.sanousun.mdweather.ui.fragment

import android.app.AlertDialog
import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.os.Bundle
import android.preference.Preference
import android.preference.PreferenceFragment

import com.sanousun.mdweather.R

import de.psdev.licensesdialog.LicensesDialog

class AboutFragment : PreferenceFragment(), Preference.OnPreferenceClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.preference)
        findPreference("license").onPreferenceClickListener = this
        findPreference("email").onPreferenceClickListener = this
        findPreference("github").onPreferenceClickListener = this
    }

    override fun onPreferenceClick(preference: Preference): Boolean {
        val cm = activity.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
        when (preference.key) {
            "license" ->
                LicensesDialog.Builder(activity)
                        .setNotices(R.raw.notices)
                        .setIncludeOwnLicense(true)
                        .build()
                        .showAppCompat()
            else ->
                AlertDialog.Builder(activity)
                        .setMessage("复制到剪贴板？")
                        .setPositiveButton("确定", { _, _ ->
                            val myClip: ClipData
                            val text = preference.summary
                            myClip = ClipData.newPlainText("text", text)
                            cm.primaryClip = myClip
                        })
                        .setNegativeButton("取消", null)
                        .show()
        }
        return true
    }

    companion object {
        fun newInstance(): AboutFragment {
            return AboutFragment()
        }
    }
}
